src/vehicle.h

```diff
@@ -416,6 +421,20 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     vehicle();
     ~vehicle () override;
 
+    /**
+     * Set stat for part constrained by range [0,durability]
+     * @note does not invoke base @ref item::on_damage callback
+     */
+    void set_hp( vehicle_part &pt, int qty );
+
+    /**
+     * Apply damage to part constrained by range [0,durability] possibly destroying it
```

Comment:
At the moment it just reduces `hp` until it reaches zero at which point the part `is_broken()`